src/MainWindow.vala

```diff
@@ -261,10 +265,10 @@ public class AppCenter.MainWindow : Gtk.ApplicationWindow {
 
     public void show_update_badge (uint updates_number) {
         if (updates_number == 0U) {
-            set_widget_visibility (updates_badge, false);
+            updates_badge_revealer.reveal_child = false;
         } else {
             updates_badge.label = updates_number.to_string ();
-            set_widget_visibility (updates_badge, true);
+            updates_badge_revealer.reveal_child = true;
         }
```

Comment:
It looks like we're only using `set_widget_visibility ()` in one other place besides this, would we want to inline that as part of this branch?

Reply:
I didn't want to touch it yet, but yeah eventually I want to get rid of that method